src/tool_storage.rs

```diff
@@ -358,6 +377,16 @@ impl ToolStorage {
         Ok(())
     }
 
+    fn trust_status(&self, name: &ToolName) -> anyhow::Result<TrustStatus> {
+        let trusted = TrustCache::read(&self.home)?;
+        let is_trusted = trusted.tools.contains(name);
+        if is_trusted {
+            Ok(TrustStatus::Trusted)
+        } else {
+            Ok(TrustStatus::NotTrusted)
+        }
+    }
```

Comment:
This method essentially does the same thing as `trust_check` did at the top before, but in the same spirit of `TrustMode` having a `TrustStatus` here instead of a `bool` makes things more explicit and allows for easier expansion of trust behavior in the future.